### DISCUSSION TITLE
IBNR - Chainladder method not working

### DISCUSSION THREAD

**Discussion by JorgenHorvei** [casact/chainladder-python#356] [Q&A] [answered]
Hi, 

I am trying to compute IBNR using the Chainladder method in Python, but have unfortunately encountered a "problem". Its actually more of an incomprehensibility rather than an actual problem. 

The case is: 
When I compute the development factors of the triangle, the numbers differ from what I would have expected them to be.

Example:
Computing the first development factor should be done by taking:  
(sum of Column 2 / sum of Column 1) - 2,063 = 2.9993586513353794

But what I actually get is: 
2.6625

Does anyone know why this is the case? I am  rather confused.
![Chainladder](https://user-images.githubusercontent.com/94543784/199459603-caca30bf-64f9-469b-a3f5-586b626b51e9.PNG)

**Comment by jbogaardt** [comment, marked as answer]
Welcome @JorgenHorvei, thank you for the question.  When `Development` is used with any of the drop features, it drops the weights assigned to specific link ratios. When the Triangle is transformed into your `transformed_triangle`, it does not alter the underlying raw data of the triangle, so your expected answer is just the volume-weighted average assuming no dropping of link ratios. Consider this re-work of your verification of the computed LDF:

```python
>>> import chainladder as cl
>>>
>>> triangle = cl.load_sample('raa')
>>> volume = cl.Development().fit_transform(triangle)
>>> dropping = cl.Development(drop=('1982', 12), drop_valuation='1988').fit_transform(triangle)
>>> # Does the underlying raw data match?:
>>> volume==dropping
True
>>> # Do the computed LDFs match?:
>>> volume.ldf_==dropping.ldf_
False
>>> # Do the link-ratios match?:
>>> volume.link_ratio==dropping.link_ratio
False
>>> # Which link ratios have been dropped?:
>>> dropping.link_ratio
         12-24     24-36     36-48     48-60     60-72     72-84     84-96    96-108   108-120
1981  1.649840  1.319023  1.082332  1.146887  1.195140  1.112972  1.033261       NaN  1.009217
1982       NaN  1.259277  1.976649  1.292143  1.131839  0.993397       NaN  1.033088       NaN
1983  2.636950  1.542816  1.163483  1.160709  1.185695       NaN  1.026374       NaN       NaN
1984  2.043324  1.364431  1.348852  1.101524       NaN  1.037726       NaN       NaN       NaN
1985  8.759158  1.655619  1.399912       NaN  1.008669       NaN       NaN       NaN       NaN
1986  4.259749  1.815671       NaN  1.225512       NaN       NaN       NaN       NaN       NaN
1987  7.217235       NaN  1.124977       NaN       NaN       NaN       NaN       NaN       NaN
1988       NaN  1.887433       NaN       NaN       NaN       NaN       NaN       NaN       NaN
1989  1.721992       NaN       NaN       NaN       NaN       NaN       NaN       NaN       NaN
>>> # Manually compute:
>>> expected = ((triangle.loc[..., 24].sum() - 4285 - 6947) /
...             (triangle.loc[..., 12].sum() - 106 - 1351 - 2063))
>>> actual = dropping.ldf_.iloc[:, :, 0, 0].sum()
>>> # Can I manually compute the link ratios?:
>>> actual == expected
True
```

**Reply by JorgenHorvei** [reply]
Thank you so much @jbogaardt! This was very helpful and it solved my problem/confusion.
